### ISSUE TITLE
Possible issues upgrading from beta2 to rc3

### ISSUE THREAD

**Issue by fantix** [edgedb/edgedb-cli#587]
- EdgeDB Version: -
- EdgeDB CLI Version: 1.0-rc3
- OS Version: ArchLinux

1. (start from 1.0-beta2)
```console
[fantix@fantix-arch ~]$ edgedb
[2021-12-01T14:50:53Z WARN  edgedb::version_check] Newer version of edgedb tool exists 1.0.0-rc.3 (current 1.0.0-beta.2). To upgrade run `edgedb self-upgrade`
...
```
2. (follow the help message to continue)
```console
[fantix@fantix-arch ~]$ edgedb self-upgrade
[2021-12-01T14:51:11Z WARN  edgedb::version_check] Newer version of edgedb tool exists 1.0.0-rc.3 (current 1.0.0-beta.2). To upgrade run `edgedb self-upgrade`
[2021-12-01T14:51:12Z WARN  edgedb::version_check] Newer version of edgedb tool exists 1.0.0-rc.3+c7b9b9b (current 1.0.0-rc.3). To upgrade run `edgedb cli upgrade`
Upgraded to version 1.0.0-rc.3 (revision 202111301935)
[fantix@fantix-arch ~]$ edgedb
[2021-12-01T14:51:14Z WARN  edgedb::version_check] Newer version of edgedb tool exists 1.0.0-rc.3+c7b9b9b (current 1.0.0-rc.3). To upgrade run `edgedb cli upgrade`
[2021-12-01T14:51:14Z WARN  edgedb::cli::directory_check] Edgedb CLI has stopped using '/home/fantix/.edgedb' for storing data and now uses standard locations of your OS. Run `edgedb cli migrate` to update the directory layout.
...
```

3. (I should've probably had `edgedb cli migrate` ran, but I ran `edgedb cli upgrade` from the first help message)

```console
[fantix@fantix-arch ~]$ RUST_LOG=info edgedb cli upgrade
[2021-12-01T14:53:30Z INFO  edgedb::portable::repository] Fetching JSON at https://packages.edgedb.com/archive/.jsonindexes/x86_64-unknown-linux-musl.json
[2021-12-01T14:53:30Z INFO  edgedb::portable::repository] Downloading https://packages.edgedb.com/archive/x86_64-unknown-linux-musl/edgedb-cli-1.0.0-rc.3+c7b9b9b.zst -> /home/fantix/.local/bin/edgedb.download
[2021-12-01T14:53:30Z INFO  edgedb::portable::repository] Fetching JSON at https://packages.edgedb.com/archive/x86_64-unknown-linux-musl/edgedb-cli-1.0.0-rc.3+c7b9b9b.zst
edgedb error: failed to download file at URL: https://packages.edgedb.com/archive/x86_64-unknown-linux-musl/edgedb-cli-1.0.0-rc.3+c7b9b9b.zst: writing "/home/fantix/.local/bin/edgedb.download": could not create `/home/fantix/.local/bin/edgedb.download`: No such file or directory (os error 2)
[fantix@fantix-arch ~]$ ls .local/bin
ls: cannot access '.local/bin': No such file or directory
[fantix@fantix-arch ~]$ edgedb --version
EdgeDB CLI 1.0.0-rc.3
```

4. (create `~/.local/bin` to continue, note that some color is wrong)

![image](https://user-images.githubusercontent.com/1751601/144260268-269da648-5b2f-48e6-8ca2-01baade8473e.png)

5. (after `source ~/.bash_profile`, all is back to normal. Portable builds are working fine.)

```console
[fantix@fantix-arch ~]$ edgedb --version
EdgeDB CLI 1.0.0-rc.3+c7b9b9b
```

**Comment by tailhook** [comment]
(5) is a known old issue. We didn't come up with solution that works for all shells, but basically it's irrelevant anymore.

Warning on (2) is fixed in #586. And (4) is fixed by #589. Anything left?

**Comment by fantix** [comment]
Awesome! If I ran `edgedb cli migrate` at (3), I think the CLI will mkdir `~/.local/bin` for me. So that's all I believe.
